site/content/entry/accurate-audio-emulation.md

```diff
@@ -0,0 +1,240 @@
+With the simple approach in place, data was being transferred into the emulated DSP, but the output was always zero.
+Debug the code as he might, it was just too frustrating to figure out what was wrong.
+Is there an error in the interpreter? MMIO? Audio Output? Maybe it's actually in DMA?
+At this point, the stress was too much to handle.
+The fear of failure after spending so much time and effort on this project really sunk in, and [wwylele](https://github.com/wwylele) just stopped working on Teakra.
+By the middle of September, [wwylele](https://github.com/wwylele) had just had enough, and decided to work on something else in the mean time.
+
+{{< figure src="/images/entry/accurate-audio-emulation/conversation.png" 
+    title="It turns out developers are still just regular human beings. Thankfully everything turned out okay!" 
+>}}
+
+## Success... Finally!
+
+The end of November rolls around with no news about DSP LLE, and [wwylele](https://github.com/wwylele) is working tirelessly to fix many other core issues, including [fixing issues with how Citra handled memory that prevented many cheat codes from working properly](https://github.com/citra-emu/citra/pull/4392) (but more on that in another blog post!)
+Out of nowhere, there's new progress on DMA.
+[wwylele](https://github.com/wwylele) worked out some extra details about the different patterns that the DMA can use to copy memory and things started taking off from there.
+The next day, he noticed an [oversight](https://github.com/wwylele/teakra/commit/d22c2d99d3b0858eb190bf9298815fcac237793a#diff-fe5ba875bf987b9df6685326294fe86fR13) in his simple DMA approach, which ended up causing the DMA to copy data to the *program* region in memory instead of the data region!
+Now that data was going to the right spot, it was much easier to debug.
+
+After some more reverse engineering and hardware testing, wwylele had a good idea of how to recreate DMA completely now, and tried it out on a custom application that produces just a simple sine wave.
+
+{{< audio src="/resources/entry/accurate-audio-emulation/sine_wav_deformed.ogg" >}}
+
+After some more tinkering, and with renewed enthusiasm, he started finding and fixing minor bugs here and there.
+All of the sudden, everything just started working.
+
+{{< audio src="/resources/entry/accurate-audio-emulation/kirby_bad.ogg" >}}
+
+And with a few more fixes...
+
+{{< audio src="/resources/entry/accurate-audio-emulation/kirby_good.ogg" >}}
+
+Finally, on Dec 6th, this is what happened after we hooked it into Citra's audio framework (sped up to full speed.)
+
+{{< audio src="/resources/entry/accurate-audio-emulation/pokemonx.ogg" >}}
+
+{{< figure src="/images/entry/accurate-audio-emulation/mhgen_najarala_armor_4K.png"
+    title="The audio used to be too loud after changing armor in Monster Hunter Generations, but this bug is no more!"
```

Comment:
nit: changing **to certain** armor

Previous wording sounds like it would happen with any armor. Ignore if being too specific is unnecessary.